test/e2e/deployment/rbac.go

```diff
@@ -0,0 +1,147 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployment
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	log "github.com/sirupsen/logrus"
+	"github.com/soda-cdm/kahu/client"
+	"github.com/soda-cdm/kahu/controllers/app/options"
+	k8s "github.com/soda-cdm/kahu/test/e2e/util/k8s"
+	kahu "github.com/soda-cdm/kahu/test/e2e/util/kahu"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//testcase for E2E deployment backup and restore
+var _ = Describe("rbacBackup", Label("rbac"), func() {
+	Describe("rbacE2ETest", func() {
+		Context("Create backup of rbac and restore", func() {
+			It("rbac", func() {
+
+				optManager, err := options.NewOptionsManager()
+				Expect(err).To(BeNil())
+				if err != nil {
+					log.Fatalf("Failed to initialize controller option manager")
+				}
+				cfg, err := optManager.Config()
+				Expect(err).To(BeNil())
+				if err != nil {
+					log.Errorf("Failed to get configuration %s", err)
+					os.Exit(1)
+				}
+
+				f := client.NewFactory(AgentBaseName, &cfg.KahuClientConfig)
+				kubeClient, err := f.KubeClient()
+				kahuClient, err := f.KahuClient()
+
+				//Create rbac to test
+				ns := "default"
+				ctx := context.TODO()
+				UUIDgen, _ := uuid.NewRandom()
+				clusterRolename := "clusterrole" + "-" + UUIDgen.String()
+				serviceAccount := "serviceaccount" + "-" + UUIDgen.String()
+				clusterrolebinding := "clusterrolebinding" + "-" + UUIDgen.String()
+				podName := "pod" + "-" + UUIDgen.String()
+
+				//create serviceAccount
+				err = k8s.CreateServiceAccount(ctx, kubeClient, ns, serviceAccount)
+				Expect(err).To(BeNil())
+
+				//create pod with the serviceAccount
+				pod, err := k8s.CreatePodWithServiceAccount(kubeClient, ns, podName, serviceAccount)
+				log.Infof("pod with sa is %v\n", pod)
+				Expect(err).To(BeNil())
+
+				err = k8s.CreateRBACWithBindingSA(ctx, kubeClient, ns, serviceAccount, clusterRolename, clusterrolebinding)
+				Expect(err).To(BeNil())
+
+				_, err = k8s.GetClusterRoleBinding(ctx, kubeClient, clusterrolebinding)
+				Expect(err).To(BeNil())
+
+				//create backup for the rbac
+				backupName := "backup" + "rbac" + "-" + UUIDgen.String()
+				includeNs := "default"
+				resourceType := "Pod"
+				backup := kahu.NewBackup(backupName, includeNs, resourceType)
+				opts := metav1.CreateOptions{}
+
+				_, err = kahuClient.KahuV1beta1().Backups().Create(ctx, backup, opts)
+				Expect(err).To(BeNil())
+				log.Infof("backup of rbac is done\n")
+				time.Sleep(40 * time.Second)
```

Comment:
I believe, Wait/poll for backup to finish is little better than sleep

Reply:
done